pkg/logs/client/tcp/destination_test.go

```diff
@@ -30,3 +34,80 @@ func TestDestinationHA(t *testing.T) {
 		assert.Equal(t, isEndpointMRF, isDestMRF)
 	}
 }
+
+// TestConnecitivityDiagnoseNoBlock ensures the connectivity diagnose doesn't
+// block
+func TestConnecitivityDiagnoseNoBlock(t *testing.T) {
+	endpoint := config.NewEndpoint("00000000", "host", 0, true)
+	done := make(chan struct{})
+
+	go func() {
+		CheckConnectivityDiagnose(endpoint)
+		close(done)
+	}()
+
+	select {
+	case <-done:
+	case <-time.After(10 * time.Second):
+		t.Error("TCP diagnosis check blocked for too long.")
+	}
+}
+
+func StartTestTCPServer(t *testing.T) (string, func()) {
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	assert.Nil(t, err)
+
+	go func() {
+		for {
+			conn, err := listener.Accept()
+			if err != nil {
+				if _, ok := err.(net.Error); ok {
+					continue
+				}
+				break // Exit the loop if the listener is closed
+			}
+
+			defer conn.Close()
+		}
+	}()
+
+	// Return the server address and a cleanup function
+	return listener.Addr().String(), func() {
+		_ = listener.Close()
+	}
+}
+
+// TestConnectivityDiagnoseFails ensures the connectivity diagnosis operates
+// correctly
+func TestConnectivityDiagnoseOperation(t *testing.T) {
```

Comment:
I think it may be clearer to split this test up and have one test for success and one test for failure. That would allow each behavior to be tested and reported independently.